devops/scripts/benchmarks/html/chart-annotations.js

```diff
@@ -0,0 +1,207 @@
+// Copyright (C) 2025 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+/**
+ * Find version changes in data points to create annotations
+ * @param {Array} points - Data points to analyze
+ * @param {string} versionKey - Key to track version changes
+ * @returns {Array} - List of change points
+ */
+function findVersionChanges(points, versionKey) {
+    if (!points || points.length < 2) return [];
+
+    const changes = [];
+    // Sort points by date
+    const sortedPoints = [...points].sort((a, b) => a.x - b.x);
+    let lastVersion = sortedPoints[0][versionKey];
+
+    for (let i = 1; i < sortedPoints.length; i++) {
+        const currentPoint = sortedPoints[i];
+
+        const currentVersion = currentPoint[versionKey];
+        if (currentVersion && currentVersion !== lastVersion) {
+            changes.push({
+                date: currentPoint.x,
+                newVersion: currentVersion,
+            });
+            lastVersion = currentVersion;
+        }
+    }
+    
+    return changes;
+}
+
+/**
+ * Add version change annotations to chart options
+ * @param {Object} data - Chart data
+ * @param {Object} options - Chart.js options object
+ */
+function addVersionChangeAnnotations(data, options) {
+    const changeTrackers = [
+        {
+            // Benchmark repos updates
+            versionKey: 'gitBenchHash',
+            sources: [
+                {
+                    name: "Compute Benchmarks",
+                    url: "https://github.com/intel/compute-benchmarks.git",
+                    color: {
+                        border: 'rgba(220, 53, 69, 0.8)',
```

Comment:
I know it's a minor thing, but maybe we should pick colors that aren't similar to what we use for data series? someone might misinterpret the annotation as relating to just one series.
(i'm not sure it's going to be possible).

Reply:
ok, I think I preferred the predefined colors, but pick a pastel palette so that it's differentiated from the series colors.

Reply:
the colors has changed, an example:
![image](https://github.com/user-attachments/assets/0b310adb-242c-4d0f-b654-77f8ff179acb)